doc/UI-Tesztek.md

```diff
@@ -0,0 +1,34 @@
+# UI Tesztek Selenium-mal
+
+## Technológia
+A felhasználói felületet Selenium könyvtárral Python 3.9-es környezetben készítettem. A tesztek célja, hogy vizuálisan szemléltesse a felhasználói felület használatát hasonlóan egy emberi felhasználó kattintásaihoz. Ennél többet is tudnak azonban a tesztek, mivel segédfüggvények és "assertion"-ök segítségével kódból tesztelhetőek az elvárt kimenetek, pl. szövegek, elvárt listaelemek stb. Ha ezek nem felelnek meg az elvárásnak, a script megszakítja a tesztet és logolja a felismert hibát, így a program későbbi módosításai esetén felismeri az UI-n történt nem várt változásokat.
+
+## A program futtatása és szükséges kiegészítők
+A programot a main.py elindításával lehet használni. A script elindítja a backend-et és a frontendet is, ezért ezeket külön nem kell indítanunk (nem is szabad, mert ekkor többen próbálnák ugyanazt a portot használni!).
+Ekkor megnyílik egy Chrome böngésző, amelyen a script gombnyomásokat, szövegek beírását hajtja végig, így láthatjuk a tesztelt folyamatok vizuális lefutását, illetve a program a fontosabb vizsgálatoknál logolja az eredményeket.
+A futás végén a konzolon feltett _Terminate batch job?_ kérdésre "y"-vel válaszoljunk, ez állítja le a frontendet.
+
+_Szükséges függőségek_: 
```

Comment:
A pip install selenium kimaradt
Ezt érdemes egy külön requirements.txt-be rakni